src/crewai/memory/long_term/long_term_memory.py

```diff
@@ -14,14 +15,67 @@ class LongTermMemory(Memory):
     LongTermMemoryItem instances.
     """
 
-    def __init__(self, storage=None, path=None):
+    def __init__(
+        self, 
+        storage=None, 
+        storage_type: str = "sqlite",
+        path: Optional[str] = None,
+        postgres_connection_string: Optional[str] = None,
+        postgres_schema: Optional[str] = None,
+        postgres_table_name: Optional[str] = None,
+        postgres_min_pool_size: Optional[int] = None,
+        postgres_max_pool_size: Optional[int] = None,
+        postgres_use_connection_pool: Optional[bool] = None,
+    ):
```

Comment:
I like that `storage_type` can be used as a factory parameter. However, I think this class should remain storage-type agnostic. We could rely on `**kwargs` instead of mapping all supported attributes for each storage type here. IMO, the factory (as the PostgresStorageFactory) should be responsible for handling those mappings.. not this class